packages/tx/src/eip4844Transaction.ts

```diff
@@ -0,0 +1,479 @@
+   */
+  public static fromSerializedTx(serialized: Buffer, opts?: TxOptions) {
+    const decoded = SignedBlobTransactionType.deserialize(serialized.slice(1))
+    const tx = decoded.message
+    const accessList: AccessListBuffer = []
+    for (const listItem of tx.accessList) {
+      const address = Buffer.from(listItem.address)
+      const storageKeys = listItem.storageKeys.map((key) => Buffer.from(key))
+      const accessListItem: AccessListBufferItem = [address, storageKeys]
+      accessList.push(accessListItem)
+    }
+    const to = tx.to.value === null ? undefined : Address.fromString(toHexString(tx.to.value))
+    const versionedHashes = tx.blobVersionedHashes.map((el) => Buffer.from(el))
+    const txData = {
+      ...tx,
+      ...{
+        versionedHashes,
+        to,
+        accessList,
+        r: decoded.signature.r,
+        s: decoded.signature.s,
+        v: BigInt(decoded.signature.yParity),
+        gasLimit: decoded.message.gas,
+      },
+    } as BlobEIP4844TxData
+    return new BlobEIP4844Transaction(txData, opts)
+  }
+
+  getUpfrontCost(baseFee: bigint = BigInt(0)): bigint {
+    const prio = this.maxPriorityFeePerGas
+    const maxBase = this.maxFeePerGas - baseFee
+    const inclusionFeePerGas = prio < maxBase ? prio : maxBase
+    const gasPrice = inclusionFeePerGas + baseFee
+    return this.gasLimit * gasPrice + this.value
+  }
+
+  raw(): TxValuesArray {
+    throw new Error('Method not implemented.')
+  }
+
+  /**
+   * Serialize a blob transaction to the execution payload variant
+   * @returns the minimum (execution payload) serialization of a signed transaction
+   */
+  serialize(): Buffer {
+    const to = {
+      selector: this.to !== undefined ? 1 : 0,
+      value: this.to?.toBuffer() ?? null,
+    }
+    const sszEncodedTx = SignedBlobTransactionType.serialize({
+      message: {
+        chainId: this.common.chainId(),
+        nonce: this.nonce,
+        maxPriorityFeePerGas: this.maxPriorityFeePerGas,
+        maxFeePerGas: this.maxFeePerGas,
+        gas: this.gasLimit,
+        to,
+        value: this.value,
+        data: this.data,
+        accessList: this.accessList.map((listItem) => {
+          return { address: listItem[0], storageKeys: listItem[1] }
+        }),
+        blobVersionedHashes: this.versionedHashes,
+        maxFeePerDataGas: this.maxFeePerDataGas,
+      },
+      // TODO: Decide how to serialize an unsigned transaction
+      signature: {
+        r: this.r ?? BigInt(0),
+        s: this.s ?? BigInt(0),
+        yParity: this.v === BigInt(1) ? true : false,
+      },
+    })
+    return Buffer.concat([TRANSACTION_TYPE_BUFFER, sszEncodedTx])
+  }
+
+  serializeNetworkWrapper(): Buffer {
+    const to = {
+      selector: this.to !== undefined ? 1 : 0,
+      value: this.to?.toBuffer() ?? null,
+    }
+    const blobArrays = this.blobs?.map((blob) => Uint8Array.from(blob)) ?? []
+    const serializedTxWrapper = BlobNetworkTransactionWrapper.serialize({
```

Comment:
I 100% not understand why this `BlobNetworkTransactionWrapper` has a method `serialize()`. 😋

Reply:
The main place this would be used in is devp2p transaction gossip.  When blob transactions are gossipped around the mem pool, they should be in the network wrapper format.

Reply:
Ha, no, I mean: where is this method "coming from"? 😋 I haven't found it in the code base, is this somehow coming from the new `ContainerType` in `types.ts`? 🤔

Also: `BlobNetworkTransactionWrapper` or `BlobTransactionNetworkWrapper`? 😋